### ISSUE TITLE
TFLite build script fails

### ISSUE THREAD

**Issue by 25pwn** [google/ml-compiler-opt#292]
I'm trying to build TFLite on Fedora 38 using `buildbot/build_tflite.sh`. All prior projects build, but TFLite fails to build with the following:
```
[21/210] Building CXX object CMakeFiles/tensorflow-lite.dir/kernels/internal/spectrogram.cc.o
FAILED: CMakeFiles/tensorflow-lite.dir/kernels/internal/spectrogram.cc.o 
/usr/bin/clang++ -DCPUINFO_SUPPORTED_PLATFORM=1 -I/home/fedora/ml-opt/tflite/tensorflow/src/tensorflow -I/home/fedora/ml-opt/tflite/tensorflow/src/tensorflow-build/farmhash/src -I/home/fedora/ml-opt/tflite/tensorflow/src/tensorflow-build/gemmlowp -isystem /home/fedora/ml-opt/tflite/eigen/include/eigen3 -isystem /home/fedora/ml-opt/tflite/ARM_NEON_2_x86_SSE/include -isystem /home/fedora/ml-opt/tflite/abseil-cpp/include -isystem /home/fedora/ml-opt/tflite/flatbuffers/include -isystem /home/fedora/ml-opt/tflite/ruy/include -isystem /home/fedora/ml-opt/tflite/cpuinfo/include -O3 -DNDEBUG -std=gnu++17 -fPIC -DTFL_STATIC_LIBRARY_BUILD -Wno-deprecated-declarations -MD -MT CMakeFiles/tensorflow-lite.dir/kernels/internal/spectrogram.cc.o -MF CMakeFiles/tensorflow-lite.dir/kernels/internal/spectrogram.cc.o.d -o CMakeFiles/tensorflow-lite.dir/kernels/internal/spectrogram.cc.o -c /home/fedora/ml-opt/tflite/tensorflow/src/tensorflow/tensorflow/lite/kernels/internal/spectrogram.cc
/home/fedora/ml-opt/tflite/tensorflow/src/tensorflow/tensorflow/lite/kernels/internal/spectrogram.cc:46:22: error: unknown type name 'uint32_t'
inline int Log2Floor(uint32_t n) {
                     ^
/home/fedora/ml-opt/tflite/tensorflow/src/tensorflow/tensorflow/lite/kernels/internal/spectrogram.cc:49:3: error: unknown type name 'uint32_t'
  uint32_t value = n;
  ^
/home/fedora/ml-opt/tflite/tensorflow/src/tensorflow/tensorflow/lite/kernels/internal/spectrogram.cc:52:5: error: unknown type name 'uint32_t'
    uint32_t x = value >> shift;
    ^
/home/fedora/ml-opt/tflite/tensorflow/src/tensorflow/tensorflow/lite/kernels/internal/spectrogram.cc:61:24: error: unknown type name 'uint32_t'
inline int Log2Ceiling(uint32_t n) {
                       ^
/home/fedora/ml-opt/tflite/tensorflow/src/tensorflow/tensorflow/lite/kernels/internal/spectrogram.cc:69:8: error: unknown type name 'uint32_t'
inline uint32_t NextPowerOfTwo(uint32_t value) {
       ^
/home/fedora/ml-opt/tflite/tensorflow/src/tensorflow/tensorflow/lite/kernels/internal/spectrogram.cc:69:32: error: unknown type name 'uint32_t'
inline uint32_t NextPowerOfTwo(uint32_t value) {
                               ^
6 errors generated.
```

**Comment by mtrofin** [comment]
+ @petrhosek 

Can you check your packages match those we install in `buildbot/buildbot_init.sh` - my hunch is this has to do with some standard include.

**Comment by mtrofin** [comment]
Did you install the packages in the `buildbot/buildbot_init.sh`. Like the stuff from line 58 down:

```
     apt-get install -qq -y \
        python3-distutils \
        python-is-python3 \
        $TF_API_DEP_PACKAGES \
        $ADMIN_PACKAGES \
        g++ \
        cmake/bullseye-backports \
        cmake-data/bullseye-backports \
        ccache \
        binutils-gold \
        binutils-dev \
        ninja-build \
...
```

**Comment by boomanaiden154** [comment]
Confirming that I can replicate this on ubuntu 23.10. Will do some more digging to see exactly what's going on. Something must've changed in something external to the project/tensorflow that needs to be handled.

**Comment by boomanaiden154** [comment]
Fixed in upstream in https://github.com/tensorflow/tensorflow/commit/9dbee4329ffd0474b939927f6d337959fb72318a. Probably some of the includes above got shuffled around or some compiler-provided header had something added or removed. I'm seeing some other errors reported with gcc-13 and integer types suddenly becoming undefined. Will open a PR bumping the versions.

**Comment by boomanaiden154** [comment]
Reopening this as the PR fixing this got reverted.

**Comment by boomanaiden154** [comment]
It shouldn't work on either of them right now as the in-tree version of `/buildbot/build_tflite.sh` hasn't changed (in terms of content) since this issue has been opened. It should work on both if you pull the patch from one of my closed PRs into your `ml-compiler-opt` tree and add the patches from the two linked PRs into your Tensorflow tree, however. Once everything lands, if there are still more build failures, please open a separate issue with the new logs.

**Comment by boomanaiden154** [comment]
It's probably a difference between Ubuntu 23.10 and Ubuntu 23.04. Your second comment mentions using 23.10 rather than 23.04. I was able to reproduce using 23.10 but never tested anything on 23.04.

**Comment by boomanaiden154** [comment]
Should be fixed in 46fa121bbdecb6876993b6237efd53f9126c871e.
